openstack/02a-MAClow/IEEE802154E.c

```diff
@@ -997,7 +1012,11 @@ port_INLINE void activity_ti2() {
    packetfunctions_reserveFooterSize(&ieee154e_vars.localCopyForTransmission, 2);
    
    // calculate the frequency to transmit on
-   ieee154e_vars.freq = calculateFrequency(schedule_getChannelOffset()); 
+   if (ieee154e_vars.isSpouriousEB == TRUE) {
+      ieee154e_vars.freq = calculateFrequency(SCHEDULE_MINIMAL_6TISCH_CHANNELOFFSET); 

```

Comment:
The channel offset used for sending EB is SCHEDULE_MINIMAL_6TISCH_CHANNELOFFSET, since EBs have to be sent also in the shared cell of minimal.
It has been added a check in schedule_addActiveSlot: if the cell going to be added is on SCHEDULE_MINIMAL_6TISCH_CHANNELOFFSET and set as TX or RX, it will not be added.